beta-20170105.md

```diff
@@ -0,0 +1,94 @@
+---
+title: What's New in beta-20170105
+toc: false
+summary: Additions and changes in CockroachDB version beta-20170105.
+---
+
+## Jan 5, 2017
+
+### Binaries
+
+<div id="os-tabs" class="clearfix">
+    <a href="https://binaries.cockroachdb.com/cockroach-beta-20170105.darwin-10.9-amd64.tgz"><button id="mac" data-eventcategory="mac-binary-release-notes">Mac</button></a>
+    <a href="https://binaries.cockroachdb.com/cockroach-beta-20170105.linux-amd64.tgz"><button id="linux" data-eventcategory="linux-binary-release-notes">Linux</button></a>
+</div>
+
+### General Changes
+
+- CockroachDB now uses a more recent version of GRPC. [#9697](https://github.com/cockroachdb/cockroach/pull/9697)
+
+### SQL Language Changes
+
+- `FILTER` clauses for aggregate functions are now supported. [#10146](https://github.com/cockroachdb/cockroach/pull/10146)
+- Columns can now be created with collated string types, although these columns cannot yet be indexed or part of the primary key. [#12294](https://github.com/cockroachdb/cockroach/pull/12294)
+- `INTERVAL` values can now be added to `DATE` values. [#12428](https://github.com/cockroachdb/cockroach/pull/12428)
+- `TIMESTAMP` values can now be compared to `DATE` values. [#12431](https://github.com/cockroachdb/cockroach/pull/12431)
+- `extract()` now supports `DATE` values. As a consequence, `extract()` no longer supports `STRING` values, which must now be explicitly cast to `TIMESTAMP` or `DATE`. [#12479](https://github.com/cockroachdb/cockroach/pull/12479)
+- `DATE` literals in views now work correctly. [#12450](https://github.com/cockroachdb/cockroach/pull/12450)
+- Parsing of `INTERVAL` values is now more consistent with PostgreSQL. [#12505](https://github.com/cockroachdb/cockroach/pull/12505) [#12559](https://github.com/cockroachdb/cockroach/pull/12559) [#12566](https://github.com/cockroachdb/cockroach/pull/12566)
+- A new aggregate function `sum_int()` is now available. It is equivalent to `sum()` but its result is type `INTEGER` instead of `DECIMAL`.
```

Comment:
To be concrete, the idea @knz is proposing here is to add another `sum` overload with the signature `sum(int)->int` (in addition to `sum(int)->decimal`) that aggregates in an `int64`. While overloads with the same argument types but different return types does not make sense in typical languages where overloads are uniquely determined from their arguments, our use of demand-driven overload resolution means that this new overload could be useful in cases where an `int` return type is desired (ie. `sum():::int`). To avoid ambiguity in the non-demanded case, we would use our "preferred overload" mechanism to prefer the standard `sum(int)->decimal` implementation.

While this approach would work in practice, it would be a deviation from our previous convention of uniquely identifying overloads by their argument types. As such, it deserves discussion. @knz do you mind opening an issue to discuss the idea? We might also want to hold off on the inclusion of this new function in the release notes until this question is resolved.

@RaduBerinde